lib/ladybug/energy_model/aperture.rb

```diff
@@ -0,0 +1,110 @@
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'ladybug/energy_model/model_object'
+
+require 'json-schema'
+require 'json'
+require 'openstudio'
+
+module Ladybug
+  module EnergyModel
+    class Aperture < ModelObject
+      attr_reader :errors, :warnings
+
+      def initialize(hash)
+        super(hash)
+        raise "Incorrect model type '#{@type}'" unless @type == 'Aperture'
+      end
+
+      def defaults
+        result = {}
+        result
+      end
+
+      def find_existing_openstudio_object(openstudio_model)
+        object = openstudio_model.getSubSurfaceByName(@hash[:name])
+        return object.get if object.is_initialized
+        nil
+      end
+
+      def create_openstudio_object(openstudio_model)
+        openstudio_vertices = OpenStudio::Point3dVector.new
+        @hash[:geometry][:boundary].each do |vertex|
+          openstudio_vertices << OpenStudio::Point3d.new(vertex[0], vertex[1], vertex[2])
+        end
+
+        openstudio_construction = nil
+        if @hash[:properties][:energy][:construction]
+          construction_name = @hash[:properties][:energy][:construction]
+          construction = openstudio_model.getConstructionByName(construction_name)
+          unless construction.empty?
+            openstudio_construction = construction.get
+          end
+        end
+
+        openstudio_subsurface = OpenStudio::Model::SubSurface.new(openstudio_vertices, openstudio_model)
+        openstudio_subsurface.setName(@hash[:name])
+        openstudio_subsurface.setConstruction(openstudio_construction) if openstudio_construction
+        
+        if @hash[:boundary_condition][:type] == 'Surface'
+          openstudio_subsurface.setAdjacentSurace(@hash[:boundary_condition][:boundary_condition_objects][2])
+        end
+              
+        if @hash[:indoor_shades]
+          @hash[:indoor_shades].each do |indoor_shade|
```

Comment:
Indoor shades have no bearing on the final E+ simulation since a `ZoneSurface:Detailed` on the inside of a zone does not influence the calculation at all.  Therefore, all references to `indoor_shades` across the whole measure should be removed and we will reserve the use of indoor shades for Radiance calculations (including daylight and thermal comfort).